File: apzkr-pzpi-21-5-ivakin-nikita/Task1-Server/src/main/java/nikita/ivakin/apzpzpi215ivakinnikitatask2/model/dto/commanders/PlatCommanderDTO.java
package nikita.ivakin.apzpzpi215ivakinnikitatask2.model.dto.commanders;

import lombok.Builder;
import lombok.Data;
import nikita.ivakin.apzpzpi215ivakinnikitatask2.model.dto.groups.PlatGroupDTO;
import nikita.ivakin.apzpzpi215ivakinnikitatask2.model.enums.POST;
import nikita.ivakin.apzpzpi215ivakinnikitatask2.model.enums.RANK;
import nikita.ivakin.apzpzpi215ivakinnikitatask2.model.enums.Role;

@Data
@Builder
public class PlatCommanderDTO {

    private Integer id;

    private String firstName;

    private String lastName;

    private String secondName;

    private String passportNumber;

    private RANK rank;

    private POST post;

    private Role role;

    private Integer age;

    private String email;

    private Integer platGroupId;

    private PlatGroupDTO platGroup;
}

File: apzkr-pzpi-21-5-ivakin-nikita/Task3-WebClient/src/app/services/models/plat-commander-dto.ts
/* tslint:disable */
/* eslint-disable */
import { PlatGroupDto } from '../models/plat-group-dto';
export interface PlatCommanderDto {
  age?: number;
  email?: string;
  firstName?: string;
  id?: number;
  lastName?: string;
  passportNumber?: string;
  platGroup?: PlatGroupDto;
  platGroupId?: number;
  post?: 'BRIGADE_COMMANDER' | 'REGIMENT_COMMANDER' | 'BATTALION_COMMANDER' | 'COMPANY_COMMANDER' | 'PLAT_COMMANDER' | 'LOGISTIC_COMMANDER';
  rank?: 'MAJOR_GENERAL' | 'COLONEL' | 'MAJOR' | 'LIEUTENANT_COLONEL' | 'SENIOR_LIEUTENANT' | 'JUNIOR_LIEUTENANT' | 'SCANNING_DEVICE';
  role?: 'BRIGADE_COMMANDER' | 'BATTALION_COMMANDER' | 'COMPANY_COMMANDER' | 'PLAT_COMMANDER' | 'LOGISTIC_COMMANDER' | 'SCANNING_DEVICE' | 'ADMIN';
  secondName?: string;
}
